pkg/osversion/osversion_test.go

```diff
@@ -0,0 +1,125 @@
+package osversion_test
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-psap/special-resource-operator/pkg/osversion"
+)
+
+func TestOsversion(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Osversion Suite")
+}
+
+type testCase struct {
+	rel        string
+	maj        string
+	min        string
+	out0       string
+	out1       string
+	out2       string
+	expectsErr bool
+}
+
+func (tc *testCase) String() string {
+	str := fmt.Sprintf(
+		"%s %s.%s => [%s, %s, %s]",
+		tc.rel,
+		tc.maj,
+		tc.min,
+		tc.out0,
+		tc.out1,
+		tc.out2)
+
+	if tc.expectsErr {
+		str += " (error expected)"
+	}
+
+	return str
+}
+
+var _ = Describe("OSVersion", func() {
+	Describe("RenderOperatingSystem", func() {
```

Comment:
i don't think you need another Describe here. You can just use It directly, or add a Context, if you wish